File: Sources/CodexManager/Infrastructure/FileSystemPaths.swift
import Foundation

struct FileSystemPaths {
    var applicationSupportDirectory: URL
    var accountStorePath: URL
    var settingsStorePath: URL
    var codexAuthPath: URL
    var codexConfigPath: URL

    static func live(fileManager: FileManager = .default) throws -> FileSystemPaths {
        let appSupportBase = try fileManager.url(
            for: .applicationSupportDirectory,
            in: .userDomainMask,
            appropriateFor: nil,
            create: true
        )

        let appSupportDirectory = appSupportBase.appendingPathComponent("CodexManager", isDirectory: true)
        #if os(iOS)
        let codexDirectory = appSupportDirectory.appendingPathComponent("codex", isDirectory: true)
        #else
        let homeDirectory = fileManager.homeDirectoryForCurrentUser
        let codexDirectory = homeDirectory.appendingPathComponent(".codex", isDirectory: true)
        #endif

        return FileSystemPaths(
            applicationSupportDirectory: appSupportDirectory,
            accountStorePath: appSupportDirectory.appendingPathComponent("accounts.json", isDirectory: false),
            settingsStorePath: appSupportDirectory.appendingPathComponent("settings.json", isDirectory: false),
            codexAuthPath: codexDirectory.appendingPathComponent("auth.json", isDirectory: false),
            codexConfigPath: codexDirectory.appendingPathComponent("config.toml", isDirectory: false)
        )
    }
}

File: apps/desktop/src/main/platform/macos/file-system-paths.ts
import { posix } from "node:path";
import type { App } from "electron";
import type { FileSystemPaths } from "../../repositories/file-system-paths";

export interface MacOSPathEnvironment {
  CODEX_MANAGER_ELECTRON_SMOKE_ROOT?: string;
}

export function resolveMacOSFileSystemPaths(
  electronApp: Pick<App, "getPath">,
  env: MacOSPathEnvironment = process.env
): FileSystemPaths {
  const smokeRoot = env.CODEX_MANAGER_ELECTRON_SMOKE_ROOT;
  if (smokeRoot) {
    const applicationSupportDirectory = posix.join(smokeRoot, "app-data", "CodexManager");
    const codexDirectory = posix.join(smokeRoot, "user", ".codex");
    return {
      applicationSupportDirectory,
      accountStorePath: posix.join(applicationSupportDirectory, "accounts.json"),
      settingsStorePath: posix.join(applicationSupportDirectory, "settings.json"),
      codexAuthPath: posix.join(codexDirectory, "auth.json"),
      codexConfigPath: posix.join(codexDirectory, "config.toml")
    };
  }

  const applicationSupportDirectory = posix.join(electronApp.getPath("appData"), "CodexManager");
  const codexDirectory = posix.join(electronApp.getPath("home"), ".codex");
  return {
    applicationSupportDirectory,
    accountStorePath: posix.join(applicationSupportDirectory, "accounts.json"),
    settingsStorePath: posix.join(applicationSupportDirectory, "settings.json"),
    codexAuthPath: posix.join(codexDirectory, "auth.json"),
    codexConfigPath: posix.join(codexDirectory, "config.toml")
  };
}
